javalib/src/main/scala/java/util/regex/Pattern.scala

```diff
@@ -0,0 +1,209 @@
+package java.util
+package regex
+
+import cre2._, cre2h._
+
+import scalanative.native._, stdlib._, stdio._, string._
+
+// Inspired by: https://github.com/google/re2j/blob/master/java/com/google/re2j/Pattern.java
+
+/** Pattern is implemented using Google's re2 engine https://github.com/google/re2
+ *
+ * Some expressions are not supported:
+ *  * Character Classes
+ *    * unions: [a-d[m-p]]
+ *    * intersections: [a-z&&[^aeiou]]
+ *  * predefined character classes: \h, \H, \v, \V
+ *  * java character function classes
+ *    * \p{javaLowerCase}
+ *    * \p{javaUpperCase}
+ *    * \p{javaWhitespace}
+ *    * \p{javaMirrored}
+ *  * boundary matchers: \G, \Z, \R
+ *  * possessive quantifiers
+ *    * X?+
+ *    * X*+
+ *    * X++
+ *    * X{n}+
+ *    * X{n,}+
+ *    * X{n,m}+
+ *  * lookaheads
+ *    * (?=X)
+ *    * (?!X)
+ *    * (?<=X)
+ *    * (?<!X)
+ *    * (?>X)
+ */
```

Comment:
We don't do doc comments for java library as they are never going to be seen. This should go to the `docs/` in the section about javalib.